reference/test/factorization/ic_kernels.cpp

```diff
@@ -118,6 +135,44 @@ TYPED_TEST(Ic, SetStrategy)
 }
 
 
+TYPED_TEST(Ic, UsesSyncFreeAlgorithm)
+{
+    using value_type = typename TestFixture::value_type;
+    using index_type = typename TestFixture::index_type;
+    bool contains_sparselib = false;
+    auto hooks = make_hooks(contains_sparselib);
+    this->ref->add_logger(gko::log::ProfilerHook::create_custom(
```

Comment:
nit: I think a custom logger type that only logs operations would be easier to understand.

Reply:
I change it. does that match?

Reply:
yes, thanks for that.